### PR TITLE
Zoul: power management driver for the RE-Mote revision B

### PR DESCRIPTION
**Pull request by alignan**
This PR adds support for the RE-Mote revision B power management block, allowing to:

- Measure the voltage level of external batteries connected to the RE-Mote via its nanowatt PIC
- Shutdown mode: powers down the whole board (CC2538, CC1200, sensors, components, etc), but the nanowatt PIC and the RTCC, allowing to reduce the current draw to 150nA
- Configures the shutdown mode's awake trigger: a soft shutdown using the nanowatt PIC's timer, or a hard shutdown by configuring first the RTCC, using its `INT2` trigger
- Read back the shutdown cycle counter, allowing the application space to trigger specific routines
- Retrieve the firmware version of the nanowatt PIC

A test application is included to verify the shutdown operation.

Depends on https://github.com/contiki-os/contiki/pull/1859

### CODE REVIEW

File: examples/zolertia/zoul/rev-b/test-power-mgmt.c [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+
+  /* Read the battery voltage level */
+  if(pm_get_voltage(&voltage) != PM_SUCCESS) {
+    printf("PM: error retrieving voltage\n");
+    TEST_LEDS_FAIL;
+  }
+
+  printf("PM: Voltage (raw) = %u\n", voltage);
+
+  /* Note: if running the following text while the RE-Mote is powered over USB
```
Comment (niklas88) [inline examples/zolertia/zoul/rev-b/test-power-mgmt.c]: ..the following *test*...

Comment (alignan) [inline examples/zolertia/zoul/rev-b/test-power-mgmt.c]: Fixed, thanks!

**Comment by alignan** [comment]
Arrrgh the Doxygen warning is driving me crazy xD

**Comment by MohamedSeliem** [comment]
you might need to add these lines at the end of this file "test-power-mgmt.c "
 /**
  * @}
  * @}
 */

**Comment by alignan** [comment]
You are right, I forgot about this... thanks!

**Comment by MohamedSeliem** [comment]
you are welcome.

i looked into the Doxygen build of this PR, "ASSERT: "tok==TK_WHITESPACE" in ..\src\docparser.cpp (4718)"

i think this is a bug in Doxygen when doing this in the documented code:
/**
\* \exception \c text
*/
void foo();

so in my opinion you need to  remove all the "\c" commands from your comments, and i hope it will work this time.

**Comment by alignan** [comment]
Thanks again!

**Comment by MohamedSeliem** [comment]
I'm glad it worked out!

**Comment by alignan** [comment]
Platform specific, merging :+1:

**Pull request merged**